hudi-spark-datasource/hudi-spark4.0.x/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/Spark40ParquetReader.scala

```diff
@@ -94,14 +98,14 @@ class Spark40ParquetReader(enableVectorizedReader: Boolean,
     val schemaEvolutionUtils = new ParquetSchemaEvolutionUtils(sharedConf, filePath, requiredSchema,
       partitionSchema, internalSchemaOpt)
 
-    val fileFooter = if (enableVectorizedReader) {
+    val fileFooter = repairFooterSchema(if (enableVectorizedReader) {
       // When there are vectorized reads, we can avoid reading the footer twice by reading
       // all row groups in advance and filter row groups according to filters that require
       // push down (no need to read the footer metadata again).
       ParquetFooterReader.readFooter(sharedConf, file, ParquetFooterReader.WITH_ROW_GROUPS)
     } else {
       ParquetFooterReader.readFooter(sharedConf, file, ParquetFooterReader.SKIP_ROW_GROUPS)
-    }
+    }, tableSchemaOpt)
```

Comment:
Is this change only used for vectorized reader?  Could a comment be added here to clarify?

Reply:
it's also used for repairing the filters